src/aiodynamo/client.py

```diff
@@ -417,67 +497,56 @@ def table(self, name: TableName) -> Table:
-        select: Select = Select.all_attributes
+        projection: ProjectionExpression = None,
+        select: Select = Select.all_attributes,
     ) -> AsyncIterator[Item]:
         if projection:
             select = Select.specific_attributes
         if select is Select.count:
             raise TypeError("Cannot use Select.count with query, use count instead")
 
-        expression_attribute_names = {}
-        expression_attribute_values = {}
+        params = Parameters()
 
-        projection_expression, ean = get_projection(projection)
-        expression_attribute_names.update(ean)
+        payload = {
+            "TableName": table,
+            "KeyConditionExpression": key_condition.encode(params),
+            "ScanForward": scan_forward,
+        }
 
-        builder = ConditionExpressionBuilder()
+        if projection:
+            payload["ProjectionExpression"] = projection.encode(params)
 
         if filter_expression:
-            filter_expression, ean, eav = builder.build_expression(filter_expression)
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-
-        if key_condition:
-            key_condition_expression, ean, eav = builder.build_expression(
-                key_condition, True
-            )
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-        else:
-            key_condition_expression = None
-
-        coro_func = partial(
-            self.core.query,
-            **clean(
-                TableName=table,
-                IndexName=index,
-                ScanIndexForward=scan_forward,
-                ProjectionExpression=projection_expression,
-                FilterExpression=filter_expression,
-                KeyConditionExpression=key_condition_expression,
-                ExpressionAttributeNames=expression_attribute_names,
-                ExpressionAttributeValues=py2dy(expression_attribute_values),
-                Select=select.value,
-            )
-        )
-        async for raw in unroll(
-            coro_func,
-            "ExclusiveStartKey",
-            "LastEvaluatedKey",
-            "Items",
-            py2dy(start_key) if start_key else None,
-            limit,
-            "Limit",
-        ):
-            yield dy2py(raw, self.numeric_type)
+            payload["FilterExpression"] = filter_expression.encode(params)
+
+        if start_key:
+            payload["ExclusiveStartKey"] = start_key
+        if index:
+            payload["IndexName"] = index
+        if limit:
+            payload["Limit"] = limit
+        if select:
+            payload["Select"] = select.value
+
+        payload["ExpressionAttributeNames"] = params.get_expression_names()
+        payload["ExpressionAttributeValues"] = params.get_expression_values()
+
+        got = 0
+
+        async for result in self._fast_depaginate("Query", payload):
+            for item in result["Items"]:
```

Comment:
I'm curious what the overhead of asynchronous iteration might be... 
pages are up to 1MB, and there's network overhead anyway, so async-per-page is warranted.
otoh, items are kinda small, so async-per-item may turn out to be expensive.
Can this be benchmarked? 🙏

Reply:
I do not understand what your concern is here.

Reply:
My concern is that the top-level query API may be ineffecient.

For example, imagine that items are tiny like `{"R": 42}` and thus each page is like 100,000 items.

In that case client performing `async for item in query(): pass` may hit a bottleneck of asyncio async iteration rather than parsing dynamo response.

Reply:
For example:

```py
In [1]: async def foo():
   ...:     for i in range(10**6):
   ...:         yield i
   ...:

In [3]: async def asum():
   ...:     sum = 0
   ...:     async for i in foo():
   ...:         sum += i
   ...:     return sum
   ...:
```

This, when timed, gives me 0.17s per `await asum()`, which is 0.17 usec per `i`. Perhaps that's fast enough?

Reply:
Certainly fast enough, compared to our... 10s for 35k rows... closing.